### DISCUSSION TITLE
api:json-schema:generate what are valid --operation arguments

### DISCUSSION THREAD

**Discussion by tacman** [api-platform/api-platform#2622] [Q&A]
What are valid operations?  I've tried Get and the definition names, but I'm just guessing.

```bash
bin/console api:json-schema:generate --help
Description:
  Generates the JSON Schema for a resource operation.

Usage:
  api:json-schema:generate [options] [--] <resource>

Arguments:
  resource                   The Fully Qualified Class Name (FQCN) of the resource

Options:
      --operation=OPERATION  The operation name

```

@todo: update https://api-platform.com/docs/core/json-schema/ docs when this is answered.

**Comment by mWardi** [comment]
From what i could gather, the operation name has the following format 
`_api_/[RESOURCE_URI]{._format}_[METHOD]` 

eg. for resource named Book. 

- `_api_/books{._format}_get`
- `_api_/books{._format}_post`
- `_api_/books/{id}{._format}_put`
- `_api_/books/{id}{._format}_patch`
